Functions/xASL_quant_HadamardDecoding.m

```diff
@@ -0,0 +1,165 @@
+function [imASL_reorder] = xASL_quant_HadamardDecoding(Encoded_ASL, xDecodingFields, NumberEchoTimes)
+%xASL_im_HadamardDecoding Hadamard-4 & Hadamard-8 Decoding
+%
+% FORMAT:       [Decoded_ASL] = xASL_im_HadamardDecoding(HadamardType, sec)
+%
+% INPUT:        HadamardType  - 4 for Hadamard4
+%                             - 8 for Hadamard8
+%                             - 12 for Hadamard12
+%               sec           - output from xASL_im_HadamardCaseByCase
+%
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:  Hadamard-4 & Hadamard-8 Decoding.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:      n/a
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+% Decoding Fields
+
+TimeEncodedMatrixType = xDecodingFields.TimeEncodedMatrixType;
+TimeEncodedMatrixSize = xDecodingFields.TimeEncodedMatrixSize;
+
+if isfield(xDecodingFields,'DecodingMatrix') && ~isempty(xDecodingFields.DecodingMatrix)
+    DecodingMatrix_input = xDecodingFields.DecodingMatrix;
+else
+    DecodingMatrix_input = [];
+end
+
+% #### For Walsh Decoding Matrix ####
```

Comment:
Also here, ### is the same as %%% but for a non-Matlab language ;)

Reply:
Please check `xASL_system` which I just created as a template, also for splitting the function in steps that are then easy to read, and that are listed in the function header.

Reply:
Yeah, but again, xASL_system was created in the meantime. Let's improve those in the next PR with Koens data, then we don't have to do that much cherry-picking and so on.